src/Sulu/Bundle/CategoryBundle/Category/KeyWordManager.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CategoryBundle\Category;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Bundle\CategoryBundle\Entity\Category;
+use Sulu\Bundle\CategoryBundle\Entity\KeyWord;
+
+/**
+ * Manages keyword for categories.
+ */
+class KeyWordManager implements KeyWordManagerInterface
+{
+    /**
+     * @var KeyWordRepositoryInterface
+     */
+    private $repository;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function __construct(KeyWordRepositoryInterface $repository, EntityManagerInterface $entityManager)
+    {
+        $this->repository = $repository;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(KeyWord $keyWord, Category $category)
+    {
+        if (null !== $synonym = $this->findSynonym($keyWord)) {
+            // reset entity and remove it from category
+            if ($this->entityManager->contains($keyWord)) {
+                $this->entityManager->refresh($keyWord);
+            }
+            $this->delete($keyWord, $category);
+
+            // link this synonym to the category
+            $keyWord = $synonym;
+        }
+
+        $categoryTranslation = $category->findTranslationByLocale($keyWord->getLocale());
+
+        // if key-word already exists in category
+        if ($categoryTranslation->hasKeyWord($keyWord)) {
+            return $keyWord;
+        }
+
+        $keyWord->addCategoryTranslation($categoryTranslation);
+        $categoryTranslation->addKeyWord($keyWord);
+
+        // FIXME category and meta will no be updated if only keyword was changed
```

Comment:
Is this comment still valid? If yes it should probably be `will not be updated`.